src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1763,6 +1762,11 @@ internal static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSt
                 body = null;
             }
 
+            if (method.IsConstructor() && method.IsImplicitlyDeclared && nullableInitialState is object)
+            {
+                NullableWalker.AnalyzeIfNeeded(compilationState.Compilation, method, body ?? BoundBlock.SynthesizedNoLocals(method.GetNonNullSyntaxNode()), diagnostics, useConstructorExitWarnings: true, nullableInitialState, out _);
```

Comment:
>BoundBlock.SynthesizedNoLocals(method.GetNonNullSyntaxNode()) [](start = 93, length = 61)

This is repeated several times in the class, for methods and containing type. Consider extracting a helper method.

Reply:
Is the suggestion to create a local function in here like `BoundBlock getSynthesizedEmptyBody() => BoundBlock.SynthesizedNoLocals(method.GetNonNullSyntaxNode())`?

Reply:
The suggestion was a `static BoundBlock GetSynthesizedEmptyBody(Symbol symbol) { ... }` method which could be used for methods and containing types.

---
In reply to: [468253616](https://github.com/dotnet/roslyn/pull/46441#discussion_r468253616) [](ancestors = 468253616)

Reply:
Done